lib/Memory/Store.js

```diff
@@ -85,11 +83,16 @@ class MemoryStore extends Store {
     const data = this.data.get(path);
     data.state = state;
     data.observers.forEach(([onChange]) => onChange(state));
+  }
+
+  set(query, valueOrState) {
```

Comment:
I'm not really proud of having `valueOrState` or `pathOrAction` in the code. Does this not mean the function is too much generic?